src/services/directoryServices/MediaDirectoryService.js

```diff
@@ -28,13 +28,16 @@ class MediaDirectoryService {
     return files
   }
 
-  async listFiles(sessionData, { directoryName }) {
+  async listMediaDirectoryContent(sessionData, { directoryName, page, limit }) {
     if (!isMediaPathValid({ path: directoryName }))
       throw new BadRequestError("Invalid media folder name")
 
-    return await this.gitHubService.readMediaDirectory(
+    return this.gitHubService.readMediaDirectory(
       sessionData,
-      directoryName
+      directoryName,
+      // NOTE: Done here as frontend starts indexing from 1
+      page - 1,
```

Comment:
Not a blocker but I feel that doing this might trip other engineers a little bit, because a "page 1" on the frontend is different from the backend. I have a couple of suggestions:
1. Switch to using offset instead (which seems to be recommended over using page, but I'm ok either way)
2. Bring this "minus 1" logic closer to the front, meaning the frontend's Paginator component should do this immediately on user change.
3. Bring this "minus 1" logic closer to the back, meaning this happens inside `getPaginatedDirectoryContents`.

I feel that having this in-between makes the definition of a page inconsistent and might add a little mental overhead.

Reply:
could i get clarity on what you meant by using offset? 
 
we can't do #2 because the pagination components informs us of the selected page - and we maintain this using a `curPage` state variable, which feeds back into the component itself.

my main considerations here were that the router should extract the variables from the request and defer business logic to the service (the media service here), which is why i left it here. 

idt i'll change it to being in `getPaginatedDirectoryContents` but wdyt about having it on the router level then?

Reply:
ok i derped on this - you're right wrt the subtraction! i'll put this in the FE, probably in a custom `usePaginate` hook for clarity (so it'll do the auto conversion).

Reply:
wrt offset - agreed on more ideal. the reference i was looking at (can't remember offhand) used page + limit, which i duplicated here ._. but we can swap to a offset + limit approach (as outlined [here](https://specs.openstack.org/openstack/api-wg/guidelines/pagination_filter_sort.html)). 

i might not do this as idt it impacts clarity + involves both FE and BE changes which is annoying. lmk if you have concerns though (ie, 1 is more powerful than the other for eg) and we can change from there?